library/core/src/sync/atomic.rs

```diff
@@ -1271,11 +1297,16 @@ impl AtomicBool {
     ///
     /// # Considerations
     ///
-    /// This method is not magic; it is not provided by the hardware.
-    /// It is implemented in terms of [`AtomicBool::compare_exchange_weak`], and suffers from the same drawbacks.
-    /// In particular, this method will not circumvent the [ABA Problem].
+    /// This method is not magic; it is not provided by the hardware, and does not act like a
+    /// critical section or mutex. It is implemented in terms of
+    /// [`compare_exchange`][Self::compare_exchange]\*, which is a [CAS operation], and thus
+    /// suffers from the usual drawbacks of CAS operations, in particular the [ABA problem].
+    ///
+    /// \* It is actually implemented using [`compare_exchange_weak`][Self::compare_exchange_weak],
```

Comment:
I don't think the strong vs. weak CAS is observable behavior for `fetch_update`. Given that I'm inclined to either remove this note or just mention `compare_exchange_weak` in the previous paragraph, now that it clarifies that the drawbacks are linked to the use of CAS, not whether or not the CAS is weak or not.